codecov_auth/migrations/0037_sync_user_terms_agreement.py

```diff
@@ -0,0 +1,31 @@
+# Generated by Django 4.2.2 on 2023-08-30 20:30
+
+from django.db import migrations
+
+
+def sync_agreements(apps, schema):
+    owners = apps.get_model("codecov_auth", "Owner")
+    for owner in owners.objects.all():
```

Comment:
This could take a long time to loop through all the owners in the database - I wonder if it makes sense to do this in a Django management command instead of a migration.

Also, consider using https://docs.djangoproject.com/en/4.2/ref/models/querysets/#iterator to avoid loading the full set of owner records into memory here.

Reply:
@trent-codecov What are your thoughts here?  Is it OK for the migrate pod to execute a really long process like this?

Reply:
Actually, looking a little closer at the code inside the loop - I think this could all be replaced with some SQL that should be much more efficient:

```sql
UPDATE users
SET users.terms_agreement_at = subquery.terms_agreement_at
FROM (
    SELECT
        owners.user_id, codecov_auth_ownerprofile.terms_agreement_at
    FROM owners
    INNER JOIN codecov_auth_ownerprofile
    ON codecov_auth_ownerprofile.owner_id = owners.ownerid
) subquery
WHERE subquery.user_id = users.id;
```

Look into `RiskyRunSQL` if you want to take this approach instead of the code-based approach.

Reply:
@scott-codecov  Switched to using SQL which seems like a better option for this.
Regarding `RiskyRunSQL`, my understanding is we don't need it since we're not altering the owners table right? A side effect with using `RiskyRunSQL` is that there will be an interval of time between the new code changes being deployed and whenever we manually run the SQL, during that interval data is reading/writing from users table before the data is migrated over from the ownerprofile table. This can lead to writes being overwritten and reads being wrong. There is probably a small chance for this to happen though, thoughts?

Reply:
@scott-codecov generally prefer to avoid long running migrate pods when possible but it's fine if necessary. I like your sql approach.

@JerrySentry the main point of risky sql / migrations is to allow you to deploy your code decoupled from sql that can/will lock the db. Then the infra team can run the sql after hours. 

Ideally you design your code such that the deploy with the risky sql can be done first. In the case that is not possible, we can just do the merge after hours, downside being you will need to be around to help just in case.